src/include/ipc4/notification.h

```diff
@@ -168,3 +230,22 @@ static inline void ipc4_notification_watchdog_init(struct ipc4_watchdog_timeout_
 	notif->primary.r.rsp = SOF_IPC4_MESSAGE_DIR_MSG_REQUEST;
 	notif->primary.r.msg_tgt = SOF_IPC4_MESSAGE_TARGET_FW_GEN_MSG;
 }
+
+/* Input Data Payload */
+union ipc4_resource_event_data {
+	/* Raw data */
+	uint32_t dws[6];
+} __packed __aligned(8);
```

Comment:
This should really be sized, i.e. we should use the reserved field in the parent to state the size here.

Reply:
Agreed Union here looks superficial. Was there any specific intent for it here ? If not it should go as Liam sugested